### DISCUSSION TITLE
I have realize the hover method in trait

### DISCUSSION THREAD

**Discussion by Decodetalkers** [ebkalderon/tower-lsp#350] [Q&A] [answered]
```rust
    async fn hover(&self, params: HoverParams) -> Result<Option<Hover>> 
```
I have realized this function, but my editor cannot receive it .. I cannot understand why

**Comment by Decodetalkers** [comment, marked as answer]
Finished. I should state that I have realize onhover in  initialize..

**Reply by ebkalderon** [reply]
Yes, the server must specify what capabilities it supports in the [`InitializeResult`](https://docs.rs/lsp-types/latest/lsp_types/struct.InitializeResult.html) returned by the `initialize()` method. Another option is to not specify statically and instead register for the capability dynamically using `Client::register_capability()` (assuming the client indicated it supports dynamic registration in [`InitializeParams`](https://docs.rs/lsp-types/latest/lsp_types/struct.InitializeParams.html))
